test/integration/src/test/java/io/pravega/test/integration/KeyValueTableTest.java

```diff
@@ -192,6 +261,1427 @@ public void testCreateDeleteKeyValueTable() {
+            entryKeyList.add(key);
+        }
+        log.info("entryKeyList: {}", entryKeyList);
+        for (int i = 0; i < keyArray.length; i++) {
+            assertEquals("Corresponding key for getEntryList should be as inserted",
+                    keyList.get(i), getEntryList1.get(i).getKey().getKey());
+            assertEquals("Corresponding key for entryList should be as inserted",
+                    entryKeyList.get(i), getEntryList1.get(i).getKey().getKey());
+        }
+
+        // Comparing each & every entry values of both entrylist and getEntryList.
+        for (int i = 0; i < keyArray.length; i++) {
+            assertEquals("All string values of both entryList1 and getEntryList1 should be equal",
+                    entryMap1.get(entryKeyList.get(i)), getEntryList1.get(i).getValue());
+        }
+        log.info("KVPair operations succeeds over KVTable");
+
+        // seal the stream.
+        CompletableFuture<Boolean> sealStreamStatus = controller.sealStream(newScope, streamName);
+        log.info("Sealing stream {}", streamName);
+        assertTrue(sealStreamStatus.get());
+        // delete the stream.
+        CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(newScope, streamName);
+        log.info("Deleting stream {}", streamName);
+        assertTrue(deleteStreamStatus.get());
+        // delete the  scope.
+        CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(newScope);
+        log.info("Deleting scope {}", newScope);
+        assertTrue(deleteScopeStatus.get());
+
+        log.info("Successfully completed creating Stream & KVTable with same " +
+                "name under same scope with event read write test on both entities");
+    }
+
+    /**
+     * Test creation of KVTable with zero (0) partition in their config value
+     */
+    @Test
+    @Ignore
+    public void testKVTablePartitions() {
+        KeyValueTableInfo newKVT = newKeyValueTableName();
+        log.info("KVTable name: {} to be tested for 0 (zero) partition size",
+                newKVT.getKeyValueTableName());
+        KeyValueTableConfiguration tableConfig = KeyValueTableConfiguration.builder()
+                .partitionCount(0).build();
+
+        assertTrue("The requested KVTable creation must fail",
+                this.controller.createKeyValueTable(SCOPE, newKVT.getKeyValueTableName(),
+                        tableConfig).join());
+
+        val segments = this.controller.getCurrentSegmentsForKeyValueTable(newKVT.getScope(), newKVT.getKeyValueTableName()).join();
+        log.info("Zero partition segemnts: {}", segments);
+        for (val s : segments.getSegments()) {
+            // We know there's nothing in these segments. But if the segments hadn't been created, then this will throw
+            // an exception.
+            this.tableStore.get(s.getKVTScopedName(), Collections.singletonList(new ByteArraySegment(new byte[1])), TIMEOUT);
```

Comment:
This returns a CompletableFuture. Where are you waiting for it? If you don't do that, you will never catch exceptions.

Reply:
Sure, I'll correct this part of code where ever it is used.